packages/atoms/connect/ConnectButton.tsx

```diff
@@ -0,0 +1,74 @@
+import { cn } from "@/lib/utils";
+import { useState } from "react";
+import * as React from "react";
+
+import { Button } from "../src/components/ui/button";
+import type { AtomsGlobalConfigProps } from "../types";
+
+// we accept two props: buttonText & onButtonClick
+// buttonText is for displaying buttonText
+// onButtonClick is a a method to handle onClick for button
+// it accepts another object which contains a callback in case of any error
+
+interface ConnectButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  buttonText?: string;
+  icon?: JSX.Element;
+  handleClick: () => void;
+  onSuccess?: () => void;
+  onError: () => void;
+}
+
+export function ConnectButton({
+  buttonText,
+  handleClick,
+  onSuccess,
+  onError,
+  className,
+  icon,
+}: ConnectButtonProps & AtomsGlobalConfigProps) {
+  const [isProcessing, setIsProcessing] = useState<boolean>(false);
+  const [isSuccess, setIsSuccess] = useState<boolean>(false);
+  const [errMsg, setErrMsg] = useState<string>("");
+
+  function handleSubmit(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
+    e.preventDefault();
+
+    try {
+      setIsProcessing(true);
+      setIsSuccess(true);
+      handleClick();
+    } catch (error) {
+      setIsProcessing(false);
+      setIsSuccess(false);
+      onError();
+      setErrMsg(error?.message);
+    }
+
+    setIsProcessing(false);
+
+    // if user wants to handle onSuccess inside handleClick then it makes no sense to have a separate handler
+    // otherwise only if the user explicitly passes an onSuccess handler this gets triggered
+    if (isSuccess && onSuccess) {
+      onSuccess();
+    }
+  }
+
+  return (
+    <div>
+      <Button
+        className={cn(
+          "bg-default text-default dark:text-muted dark:bg-muted relative inline-flex h-9 items-center whitespace-nowrap rounded-md px-4 py-2.5 text-sm font-medium !shadow-none transition-colors disabled:cursor-not-allowed",
+          className
+            ? className
+            : "bg-default text-default dark:text-muted dark:bg-muted relative inline-flex h-9 items-center whitespace-nowrap rounded-md px-4 py-2.5 text-sm font-medium !shadow-none transition-colors disabled:cursor-not-allowed"
+        )}
+        type="button"
+        disabled={!isProcessing}
+        onClick={(event) => handleSubmit(event)}>
+        {!!icon && icon}
```

Comment:
prefer using `Boolean` util instead of `!!` in most cases

Reply:
But  doesn't `!!` convert the value into it's boolean equivalent?

Reply:
Opinionated - Doesnt matter 
<img width="565" alt="CleanShot 2023-10-13 at 12 00 53@2x" src="https://github.com/calcom/cal.com/assets/55134778/78bb1876-8f16-4e09-9c90-c15695aa5862">

`{icon && icon}` also has the same effect as `{!!icon&&icon}`

Reply:
!! or Boolean doesn't matter, I'd advocate for !! actually as it is much simpler to error and write `new Boolean()` (which is objectively wrong) than `new !!`. 

Anyway we don't enforce little code style issues like this; that's up to the developer.

Reply:
indeed it it's very up to the developer and the team, I'm fine with both since the result is the same, it's something I used to enforce in the past in less experience team in react using non boolean to test whether a component should be rendered or not.

anyway my bad I should have precised it was just a suggestion and not a required change